symfit/core/objectives.py

```diff
@@ -142,11 +142,10 @@ def __eq__(self, other):
                     equal = other.data[key] == value
                 finally:
                     if not equal:
-                        return False
+                        break
             else:
                 return True
-        else:
-            return False
+        return False
```

Comment:
My suggestion:
```python3
if self.__class__ != other.__class__ or self.model != other.model:
    return False

for key, value in self.data.items():
    try:
        equal = np.allclose(other.data[key], value)
    except TypeError:
        equal = other.data[key] == value
    finally:
        if not equal:
            return False
return True

```

Reply:
I think it's harder to read with all the negations. I personally prefer to have if conditionals without `not` type operations.

Reply:
The difference is that I don't need the for/else.
In my example you very obviously try to find the short-circuit/easy answer: when they're not equal.

Reply:
Fair enough, and there something to be set for keeping indentation under control